src/transform.js

```diff
@@ -551,7 +551,29 @@ const transform = {
       }
 
       // TODO: maintain the position of indexers
-      elements.push(...indexers);
+      indexers.forEach(indexer => {
+        const value = indexer.typeAnnotation.typeAnnotation;
+        const key = indexer.parameters[0].typeAnnotation.typeAnnotation;
+        if (
+          t.isTSSymbolKeyword(key) ||
+          t.isTSStringKeyword(key) ||
+          t.isTSNumberKeyword(key)
+        ) {
+          elements.push(indexer);
+        } else {
+          const typeParameter = t.tsTypeParameter(key);
+          typeParameter.name = indexer.parameters[0].name;
+
+          const mappedType = {
+            type: "TSMappedType",
+            typeParameter: typeParameter,
+            typeAnnotation: value,
+            optional: true
```

Comment:
I think we should be copying the optionality of the flow type.

Reply:
The behaviour is different between Flow and TypeScript:

```
type Day = 'monday' | 'tuesday';

type DayStats = {
    [key: Day]: number;
};


const a: DayStats = {
  monday: 1
}
```

is ok for flow, but TypeScript complains about `a` missing key tuesday. What would be the Flow syntax for enforcing all keys in enum?

Reply:
What you have makes sense.  Thanks for the explanation.